scripts/zones/West_Sarutabaruta/npcs/Twinkle_Tree.lua

```diff
@@ -11,36 +11,40 @@ require("scripts/globals/npc_util")
 require("scripts/globals/quests")
 -----------------------------------
 local entity = {}
+local maxHour = 3
+local questNameProg = "QuestCatchAFallingStar_prog"
```

Comment:
These global variables are also unnecessary.

Reply:
Those aren't global variables.  They're local only to that script.  The values are used in multiple places in the code which generally dictates turning them into a variable and using that for ease of change.  Strings should almost always be created as variables, if nothing else but to reduce spelling errors.  

As a matter of optimization, strings are stored in internal memory each time a new string appears and looked up by lua to see if an exact string already exists, using that internal memory location of the string.  Assigning the string to a variable beforehand forces creation of the string in memory and creates the variable as a reference to that memory location so lua doesn't need to lookup the string, it already knows it exists and uses the reference to the string value, thus reducing time and memory caused by the script running. Lua.org's own Optimization/Performance tips states to use locals wherever possible specifically for this reason.

While this script may be small, we shouldn't be foregoing simple optimization techniques and good coding paradigms.

Reply:
Again this just isn't our code style. We can look into converting our code base in to using variables this but, but it just isn't the case as of yet.

Reply:
ideally this is handled in the new quest framework and doesn't use wonky variables like this to handle.

Reply:
Sanity check fixes added.
Removed all code from original scripts and rebuilt in the new quest framework that falls under scripts/quests/windurst/
Retested quest flow and works without issues as far as I can tell.